nbs/tsdataset.ipynb

```diff
@@ -54,7 +54,8 @@
     "#| export\n",
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #35.            df = pd.read_parquet(self.files_ds.files[idx])
</pre><p>Please use</p>

<pre class="ql-syntax" spellcheck="false">data = pd.read_parquet(self.files_ds.files[idx], columns=self.temporal_cols).to_numpy()
</pre><p>here. Reading only the columns that we'll use in the order we'll use them should be very fast.</p><p> </p><p>For the available mask, can you please add that when reading the directories? You can add a flag or similar and if the first file has the mask then add it to the temporal cols set and check that it's present in every file. That way the final dataset will already read the mask.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/1049/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Reply:
I've now added available_mask to temporal_cols if it is already present in all the dataframes - I'm assuming this is what you meant? To be clear I still call         
        _data, temporal_cols = TimeSeriesDataset._ensure_available_mask(data, temporal_cols)_
in the get_items method as in _from_data_directories()_ we're only reading the metadata, not the full dataframe.